src/prefect/orion/orchestration/rules.py

```diff
@@ -239,6 +239,34 @@ async def validate_proposed_state(
             None
         """
 
+        for validation_attempt in range(2):
+            try:
+                await self._validate_proposed_state()
+                return
+            except Exception:
+                # unset the run state in case it's been set
+                if self.initial_state is not None:
+                    initial_orm_state = db.FlowRunState(
+                        flow_run_id=self.run.id,
+                        **self.initial_state.dict(shallow=True),
+                    )
+                    self.session.add(initial_orm_state)
+                    self.run.set_state(initial_orm_state)
+                else:
+                    self.run.set_state(None)
+                continue
+            return
+
+        self.proposed_state = None
+        reason = "Error validating state"
```

Comment:
It might be nice to include the exception information as well, at least when debug is on.